ui/src/overridden/frontsite/Routes/LegacyRoute.js

```diff
@@ -0,0 +1,56 @@
+import { FrontSiteRoutes as CdsFrontSiteRoutes } from '../frontsiteUrls';
+import {
+  FrontSiteRoutes,
+  NotFound,
+} from '@inveniosoftware/react-invenio-app-ils';
+import { Route, Redirect } from 'react-router-dom';
+import React from 'react';
+import PropTypes from 'prop-types';
+import { legacyApi } from '../../../api/legacy/legacy';
+
+export const LegacyDocumentRoute = ({ ...props }) => {
+  return (
+    <Route
+      exact
+      path={CdsFrontSiteRoutes.legacyDocumentDetails}
+      component={LegacyDocumentCmp}
+    />
+  );
+};
+
+export class LegacyDocumentCmp extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      data: {},
+    };
+  }
+
+  componentDidMount() {
+    const { match } = this.props;
+    this.fetchDocumentFromLegacyRecId(match.params.legacyRecId);
```

Comment:
shouldn't we create a redux store for this, WDYT @ntarocco ?

Reply:
I think in the past we said that it makes sense to created redux store/state when it is important to share state, or maybe I don't  recall correctly?

Reply:
```suggestion
    this.fetchRecordFromLegacyRecId(match.params.legacyRecId);
```
since it can be series too